pkg/cmd/run/shared/shared.go

```diff
@@ -150,6 +150,25 @@ type RunsPayload struct {
 	WorkflowRuns []Run `json:"workflow_runs"`
 }
 
+func GetRunsWithFilter(client *api.Client, repo ghrepo.Interface, limit int, f func(Run) bool) ([]Run, error) {
```

Comment:
Nit: couldn't the caller just do the filtering on the result of GetRuns instead of having to inject a filter func?

Reply:
i have to filter in this way for another upcoming command and it seemed like a reasonable time to DRY but I do not have strong feelings either way